runtime/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonInput.kt

```diff
@@ -142,15 +149,39 @@ internal class StreamingJsonInput internal constructor(
         }
     }
 
-    override fun decodeBoolean(): Boolean = reader.takeString().run { if (configuration.strictMode) toBooleanStrict() else toBoolean() }
+    override fun decodeBoolean(): Boolean {
+        /*
+         * We prohibit non true/false boolean literals at all as it is considered way too error-prone,
+         * but allow quoted literal in relaxed mode for booleans.
+         */
+        return if (configuration.isLenient) {
+            reader.takeString().toBooleanStrict()
+        } else {
+            val string = reader.takeBooleanStringUnquoted()
```

Comment:
you can return `reader.takeBooleanStringUnquoted().toBooleanStrict()` right away without `val string`